GNUmakefile

```diff
@@ -16,7 +16,7 @@ doc:
 
 #### Unit tests and sanity check #############################################
 
-LAMBDAPI     = dune exec -- lambdapi check --lib-root lib
+LAMBDAPI     = dune exec -- lambdapi check --lib-root /tmp
```

Comment:
The problem is that we need to provide an existing directory for --lib-root even if the files are never going to be installed in this directory. This is annoying.

Reply:
Yeah, but we have to be prepared for the long run here. Moreover, this change also breaks installation of lambdapi. This means that now even an installed version needs to use `--lib-root`. I would advise you to revert that and to reinstate the `example` library.

Reply:
I don't understand. This variable is only used for compiling tests.

Reply:
There must be a library root. There is no question there. By removing the example you have also remove their installation under the system library root, and hence the creation of the system library root. That is why you do not just do arbitrary changes in the middle of a PR, this problem could have easily been overlooked: in the current state, installation is broken.

Reply:
Yes, there is a question. In some cases, I simply don't care about the library root. What is the "system library root"? The one under opam? But I don't want to pollute this directory with our tests, and I don't want `make` to compile the files in the old lib/ directory either.

Reply:
The library root is the directory under which packages must be installed. It is essential if we want to be able to provide a notion of installable packages at some further point. I already explained that in my previous PR on the file management stuff. This is why I first intended to distribute a standard library with Lambdapi, but for some reason you seem to think that is stupid.

For the record, I intended the standard library to be developed under the `lib` directory for many good reasons, one being the ability for tests to use this library. This is important for testing features such as the module and file system stuff that are not covered by the current tests. Developing a standard library anywhere else makes no sense to me because it definitely has to remain in sync with the tool.

To be clear, you should never have to care about the library root as a user. I don't understand why you are making such a fuss about it. The only point is that the library root cannot be assumed to exist prior to installation, that is why a path needs to be provided when running Lambdapi as a developer, prior to installation.

In any case, this is not the place to complain about the design I spent hours coming up with when working in a previous PR. This is completely __unrelated__ to the issues at hand, and this PR should simply __not break unrelated stuff__.

Reply:
Rodolphe, I never said it was stupid! I was thinking that it could be better to separate the development of the tool from the one of the library. This is what Agda and Lean people do. Testing external libraries when one makes changes is no problem. We already do it for generated libraries. This is also what Coq developers do.

Why did I remove the lib/ directory in my PR? Because make was failing while I was modifying the LP files. It was annoying. So, yes, it is unrelated but I don't think that make should compile any library or examples.

Now, what is broken? It works perfectly well on my machine.

Reply:
Agda and Lean have a big standard library, so this kind of makes sense. For now it would make more sense for us to have everything in one place I think.

You do not have to run `make` to compile Lambdapi since `dune exec - - lambdapi ...` takes care of that already before running. Basically, you never have to run `make` when you modify Lambdapi. And `make tests` should work as well.

What is broken is exactly what I said: the system library root is not created on install now. The only reason you cannot observe this is because you have an installed version of Lambdapi.

Reply:
Ok. I have put the lib/ dir back.